restore-sql-server-transaction-logs/Function/main.py

```diff
@@ -0,0 +1,195 @@
+import functions_framework
+import google.auth
+import google.auth.transport.requests
+import googleapiclient.discovery
+import logging
+import os
+from time import sleep
+from google.cloud import storage
+from google.cloud import error_reporting
+
+
+# Triggered by a change in a storage bucket
+@functions_framework.cloud_event
+def fn_restore_log(cloud_event):

```

Comment:
The "retry until done" behavior of this function is handled by the function itself, if I read the code correctly. What happens to retries if an instance of this function stops running for whatever reason?

Reply:
The function will not retry until done. There is a max_attempts config that tells the function the number of attempts. If the function breaks mid-way, the log backup file might be either restored to the CloudSQL instance or not, but it will not be deleted from the source bucket. This signifies that it was not processed successfully and that it should be manually re-uploaded. If the file is manually retried, it will be idempotently restored and deleted.

Reply:
Got it. Worth documenting this in the README, I think

Reply:
Added this info in the readme.

Reply:
SGTM